### ISSUE TITLE
Consider switching minimum JDK support to 1.8 and release 4.2.0

### ISSUE THREAD

**Issue by normanmaurer** [netty/netty#8259]
Opening this for discussion...

What are others thoughts of releasing netty 4.2.0.Final where we move to Java8 as minimum requirement ? This would allow us to get rid of some nasty reflection hacks and replace these with method handles etc. 

We could still support 4.1.x for a while in parallels. 

Once we start next major (a.k.a netty 5) we should consider even moving to java11 I think but thats for discussion after this one...

**Comment by normanmaurer** [comment]
And just to be clear 4.2.0 would be completely identical in terms of API etc with 4.1.0. The only difference would be the minimum java version requirement for now

**Comment by jasobrown** [comment]
+1. The minimum jdk version for cassandra is java 8, with cassandra version 3.0 and higher.

**Comment by normanmaurer** [comment]
Also @smaldini and @vietj

**Comment by ijuma** [comment]
Most open-source projects now require Java 8 so I think that's a good idea. With regards to 4.2.x versus 5.0.0, it seems like you're suggesting the former since the Netty APIs will remain the same even if some will see this as a breaking change.

A number of people in the Kafka community were opposed to bumping the Java version requirement without a major version bump in Kafka itself. We bumped Kafka's major version and then some were confused that there weren't major API changes in Kafka. So some people will be confused either way. :)

**Comment by normanmaurer** [comment]
@ijuma yeah the former... That said I want to start working on netty 5 soon as well, which will contain API breakage of course.

**Comment by mp911de** [comment]
I like the idea of having an artifact that can be used as a compatible drop-in for 4.1.x with improved performance in exchange for a higher JDK. With Lettuce requiring Java 8 anyways and the broad Java 8 requirements as of Project Reactor and Spring Framework, this should be a nice alignment.

**Comment by normanmaurer** [comment]
Yes that would be my plan

**Comment by nicktrav** [comment]
+1 from us!

**Comment by johnou** [comment]
Green from us.

**Comment by amizurov** [comment]
Cool, +1 from us.

**Comment by rkapsi** [comment]
@normanmaurer two :+1: from us

**Comment by trustin** [comment]
+1. Migrated to Java 8 loooong time ago. Just curious if we should call this new version 4.2 or 5.0. By choosing 5.0, 1) we can emit a strong signal that it's not a drop-in replacement and 2) we can finally hide the ancient 5.0.0.AlphaX releases from search.maven.org, but I'm just fine with 4.2 as well.

**Comment by lilaizhencn** [comment]
+1. expect since jdk 11

**Comment by LaoZhongGu** [comment]
+1.Good。And JDK11 coming soon。

**Comment by unlimitedsola** [comment]
+1, please :) , and JDK 11 support as well.

**Comment by wkgcass** [comment]
+1 for 8  
But most enterprise users might stick to java version <= 8 for a very long time (I guess they will not consider switching to higher version until Project Loom releases).

**Comment by doom369** [comment]
@normanmaurer is there any planned dates for this?

**Comment by normanmaurer** [comment]
Not yet

**Comment by mosesn** [comment]
+1 from us too.

**Comment by DATuan91** [comment]
+1 from us!

**Comment by pratikpparikh** [comment]
+1 from us too!

**Comment by chengenzhao** [comment]
+1 from us too

**Comment by njhill** [comment]
@normanmaurer thoughts on this given latest changes to `master`? Maybe there is still value in non-API-breaking java 8 version which can be a drop-in upgrade for most folks?

**Comment by vincentfree** [comment]
+1

**Comment by fabienrenaud** [comment]
Anything holding this up?

I am looking at some contention issues right now in `AddressResolverGroup` and would like to submit a patch to replace the synchronized map with a `ConcurrentMap#computeIfAbsent`... and I am sure there are other classes in the project that would benefit.

Let's do it!

**Comment by xxDark** [comment]
@normanmaurer any updates on dropping JDK versions less than 1.8? Would like to see Netty get rid of `java.lang.reflect.Method` usage, and switch to method handles and other non-`legacy` stuff.

**Comment by martin-traverse** [comment]
+1

We're running Java 11 in production now, with some legacy Hadoop stuff stuck on Java 8. I'm fairly sure our security team would knock down a suggestion to run anything on Java 7!

**Comment by doom369** [comment]
@normanmaurer so 3 years passed since that proposal and still no movements... Is there anything that blocks you maybe? Java 17 will be released soon and netty is not even on Java 8, not good.

**Comment by Glavo** [comment]
I believe that almost all JDK vendors have completely ended their support for Java 7 three years ago. So is there any new progress on this issue?

**Comment by normanmaurer** [comment]
We did release 4.2.x which depends on Java8 now.
